src/component/mxgraph/shape/render/utils.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2020 Bonitasoft S.A.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ShapeBpmnMarkerKind } from '../../../../model/bpmn/shape';
+
+const referenceOrderedMarkers = [
+  ShapeBpmnMarkerKind.LOOP,
+  ShapeBpmnMarkerKind.MULTI_INSTANCE_PARALLEL,
+  ShapeBpmnMarkerKind.MULTI_INSTANCE_SEQUENTIAL,
+  ShapeBpmnMarkerKind.COMPENSATION,
+  ShapeBpmnMarkerKind.EXPAND,
+  ShapeBpmnMarkerKind.ADHOC,
+];
+
+export function orderActivityMarkers(markers: string[]): string[] {
+  const orderedMarkers: string[] = [];
+
+  referenceOrderedMarkers.forEach(marker => {
+    if (markers.includes(marker)) {
+      orderedMarkers.push(marker);
+    }
+  });
+
+  // Put extra remaining at the end of the ordered markers, in the order they appear in the original array
+  markers
```

Comment:
When can we have a non-BPMN marker ?

Reply:
Vendor could do this, as Bonita does this for additional icons like the `Connector` icon for the `Tasks`.
I added this because it was easy to do, if you think this is useless or overengineering, I can remove it

Reply:
I didn't think about that. The Connector marker, is it in the same place than the others ?

Reply:
The connector icon is displayed on the top right of a task.
I mentioned this connector icon example to notice that there is a known need for extra icons (this can be managed by extending the existing shape) and probably for extra markers as well.

Reply:
I'm not sure vendors are adding new markers. But we can leave it. It doesn't cost anything :wink: